File: C++/Hard/CompileError/constrained-subsequence-sum_8.cpp
class Solution {
public:
    int constrainedSubsetSum(vector<int>& nums, int k) {
        
======  int n = nums.size();
=      int ans = INT_MIN;
        deque<int> q;
        fo(int i = 0; i < n; i++) {
            if(!q.mpy()) {
                nms[i] = nums[i] + q.fot();
           }
            ans = a(ans, nums[i]);
            while(!q.empty() && q.back() < nums[i]) {
                q.pop_back();
            }
            q.push_back(nums[i]);
            if(i >= k && q.front() == nums[i - k]) {
                q.pop_front();
            }
        }
        return ans

        int n = nums.size();
        vector<int> dp(n);
        int ans = nums[0];
        deque<int> q;
        for (int i = 0; i < n; ++i) {
            if (!q.empty()) {
                dp[i] = nums[i] + max(0, dp[q.front()]);
            } else {
                dp[i] = nums[i];
            }
            while (!q.empty() && dp[q.back()] <= dp[i]) {
                q.pop_back();
            }
            q.push_back(i);
            if (q.front() == i - k) {
                q.pop_front();
            }
            ans = max(ans, dp[i]);
        }
        return ans;
    }
};

File: Java/Hard/WrongAnswer/constrained-subsequence-sum_8.java
class Solution {
    public int constrainedSubsetSum(int[] nums, int k) {
        
        int n = nums.length;
        int[] dp = new int[n];
        int max = nums[0];
        dp[0] = nums[0];
        Deque<Integer> q = new ArrayDeque<>();
        q.add(0);
        for (int i = 1; i < n; i++) {
            dp[i] = Math.max(dp[i], nums[i]);
            if (q.size() > 0) {
                dp[i] = Math.max(dp[i], dp[q.getFirst()] + nums[i]);
            }
            while (q.size() > 0 && dp[q.getLast()] <= dp[i]) {
                q.removeLast();
            }
            q.add(i);
            if (i - q.getFirst() >= k) {
                q.removeFirst();
            }
            max = Math.max(max, dp[i]);
        }
        return max;
    }
}